File: screen Off/app/src/main/java/com/emranhss/screenoffandlock/MainActivity.java
package com.emranhss.screenoffandlock;

import android.app.admin.DevicePolicyManager;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private ComponentName cn;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        DevicePolicyManager dmp = (DevicePolicyManager) getSystemService(Context.DEVICE_POLICY_SERVICE);
        cn = new ComponentName(this, MyDeviceAdminReceiver.class);
        //Ekhane likhle app click korle screen of hoye jabe.
        toLock();
        finish();
        if (dmp.isAdminActive(cn)) {
            dmp.lockNow();
            finish();
        }

        findViewById(R.id.btnLockScreen).setOnClickListener(v -> {
            //Ekhane likhle app er vetor dhuke button click korle screen of hoye jabe.
//            toLock();
//            finish();
//            if (dmp.isAdminActive(cn)) {
//                dmp.lockNow();
//                finish();
//            }


        });

    } //oncreat end

    private void toLock() {
        Intent intent = new Intent(DevicePolicyManager.ACTION_ADD_DEVICE_ADMIN);
        intent.putExtra(DevicePolicyManager.EXTRA_DEVICE_ADMIN, cn);
        intent.putExtra(DevicePolicyManager.EXTRA_ADD_EXPLANATION, "Now");
        startActivity(intent);
    }

} //main end

File: Power Off/app/src/main/java/com/emranhss/poweroff/MainActivity.kt
package com.emranhss.poweroff

import android.app.admin.DevicePolicyManager
import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    private lateinit var cn: ComponentName
    private lateinit var dpm: DevicePolicyManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        dpm = getSystemService(Context.DEVICE_POLICY_SERVICE) as DevicePolicyManager
        cn = ComponentName(this, MyDeviceAdminReceiver::class.java)

        // click App icon then screen off
        if (dpm.isAdminActive(cn)) {
            dpm.lockNow()
            finish()
        } else {
            requestAdminPermission()
        }

        findViewById<Button>(R.id.btnLockScreen).setOnClickListener {
//            if (dpm.isAdminActive(cn)) {
//                dpm.lockNow()
//                finish()
//            } else {
//                requestAdminPermission()
//            }
        }

    }

    private fun requestAdminPermission() {
        val intent = Intent(DevicePolicyManager.ACTION_ADD_DEVICE_ADMIN)
        intent.putExtra(DevicePolicyManager.EXTRA_DEVICE_ADMIN, cn)
        intent.putExtra(
            DevicePolicyManager.EXTRA_ADD_EXPLANATION,
            "Permission needed to lock the screen"
        )
        startActivity(intent)
    }


//    override fun onCreate(savedInstanceState: Bundle?) {
//        super.onCreate(savedInstanceState)
//        enableEdgeToEdge()
//        setContentView(R.layout.activity_main)
//        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
//            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
//            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
//            insets
//        }
//    }
}